### PR TITLE
Continuation: Convert Week Goodie to Structured Template, update tests 

### PR DESCRIPTION
**Pull request by mintsoft**
Following on from @moollaza's fixes here: https://github.com/duckduckgo/zeroclickinfo-goodies/pull/950

I've wrapped the year handling up into the DatesRole so it's all together for now and fixed the issues that @jagtalon noticed

### CODE REVIEW

File: lib/DDG/Goodie/Week.pm [hunk 1]
```diff
@@ -42,44 +43,61 @@ my @months = qw/
     December
 /;
 
-handle query_raw => sub {
+handle query_lc => sub {
     return unless /^\s*
-        what(?:'?s|\sis|\swas)?\s+
-        (?:the\s+)?
-        (?:(current|(\d{1,2})(?:nd|th|rd|st)?)\s+)?
+        (?:what|when)(?:'?s|\sis|\swas)?\s
```
Comment (moollaza) [inline lib/DDG/Goodie/Week.pm]: looks like we need a `?` here so we can handle queries like "8th week of 2015"

**Comment by moollaza** [comment]
@mintsoft thanks a ton for picking this up! I'm glad to see the Dates role can now handle this!

**Comment by mintsoft** [comment]
@moollaza fixed

**Comment by moollaza** [comment]
LGTM :+1: Thanks a ton @mintsoft ! :smiley:

**Comment by moollaza** [comment]
@mintsoft @gsquire This is a long delayed notification, but this IA has been live for a few weeks (months?) now. Just wanted to say thanks to everyone involved.

https://duckduckgo.com/?q=3rd+week+of+2023&t=ffab&ia=answer

The IA page link should display soon...

**Comment by gsquire** [comment]
@moollaza I am grateful for your help in addition to the pushes from @mintsoft. Very glad to see this live.

**Pull request merged**